### PR TITLE
Error codes

### PR DESCRIPTION
**Pull request by SergioFierro**
## Ticket
- [CAS-164]
- 

## Description
- Add error codes for `TwilioException`
- Fix README reference to change iOS constant

## Commit message
- feat: Add error codes to `TwilioException`

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

**Review by kmorope — APPROVED** [review]

File: shared/src/commonMain/kotlin/com/twilio/passkeys/exception/TwilioException.kt [hunk 1]
```diff
@@ -16,23 +16,83 @@
...
  * This exception is used to encapsulate error messages with their types.
  *
+ * @property code The error code.
  * @property message The error message.
+ * @property cause The error cause.
  */
-data class TwilioException(
-  val message: String,
-) {
+sealed class TwilioException(val code: Int, override val message: String, override val cause: Throwable) : Exception(message, cause) {
```
Comment (yafuquen) [inline shared/src/commonMain/kotlin/com/twilio/passkeys/exception/TwilioException.kt:36]: Should the error documentation be included in the README?

Comment (SergioFierro) [reply shared/src/commonMain/kotlin/com/twilio/passkeys/exception/TwilioException.kt:36]: Added, please let me know what you think

File: shared/src/commonMain/kotlin/com/twilio/passkeys/exception/TwilioException.kt [hunk 2]
```diff
@@ -16,23 +16,83 @@
...
 
-internal const val INVALID_JSON_PAYLOAD_ERROR = "INVALID_JSON_PAYLOAD"
-internal const val UNKNOWN_ERROR = "UNKNOWN"
+internal const val DOM_EXCEPTION = 1001
+internal const val USER_CANCELED_EXCEPTION = 1002
+internal const val INTERRUPTED_EXCEPTION = 1003
+internal const val UNSUPPORTED_EXCEPTION = 1004
+internal const val NO_CREDENTIAL_EXCEPTION = 1005
+internal const val MISSING_ATTESTATION_OBJECT_EXCEPTION = 1006
+internal const val INVALID_JSON_PAYLOAD_EXCEPTION = 1008
```
Comment (ferazoguerrero) [inline shared/src/commonMain/kotlin/com/twilio/passkeys/exception/TwilioException.kt:25]: why not `1007`?

**Review by ferazoguerrero — APPROVED** [review]

File: shared/src/commonMain/kotlin/com/twilio/passkeys/exception/TwilioException.kt (hunk 2 above)
Comment (SergioFierro) [reply shared/src/commonMain/kotlin/com/twilio/passkeys/exception/TwilioException.kt]: Updated. Thanks for catching this, I totally missed it

**Pull request merged**